tests/grid/accessor_02.cc

```diff
@@ -0,0 +1,71 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2018 - 2019 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+
+// TriaAccessor<0,dim,spacedim> and TriaAccessor<0,1,spacedim> lacked
+// get_triangulation() member functions because they are not derived
+// from TriaAccessorBase. This led to awkward follow-up errors in
+// strange places when using them in certain contexts.
+
+#include <deal.II/base/logstream.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/tria_accessor.h>
+#include <deal.II/grid/tria_iterator.h>
+
+#include "../tests.h"
+
+using namespace dealii;
+
+
+template <int dim, int spacedim>
+void
+test()
+{
+  Triangulation<dim, spacedim> tria;
+  GridGenerator::hyper_cube(tria);
+  tria.refine_global(1);
+
+  std::map<typename Triangulation<dim, spacedim>::face_iterator, double> mymap;
```

Comment:
```suggestion
  std::map<typename Triangulation<dim, spacedim>::face_iterator, int> mymap;
```

The cell index is an integer. You'll get a different test output, though.

Reply:
I only copied the test `tests/grid/accessor_01.cc`; I did not modify anything except the iterator related stuff. Should I also switch from double to in in this case?

Reply:
Good idea.

Reply:
But separate pull request :-)

Reply:
@bangerth I have made your requested changes. I will setup the separate PR right away!